test/e2e_tests/namespace.go

```diff
@@ -0,0 +1,182 @@
+package e2e_tests
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/dana-team/platform-backend/src/utils/testutils"
+	"github.com/dana-team/platform-backend/src/utils/testutils/mocks"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"net/http"
+)
+
+var _ = Describe("Validate Namespace routes and functionality", func() {
+	var oneNamespaceName, secondNamespaceName string
+	var oneNamespace, secondNamespace corev1.Namespace
+
+	BeforeEach(func() {
+		oneNamespaceName = generateName("a-" + e2eNamespace)
+		oneNamespace = mocks.PrepareNamespace(oneNamespaceName, map[string]string{e2eLabelKey: e2eLabelValue})
+		createResource(k8sClient, &oneNamespace)
+
+		secondNamespaceName = generateName("b-" + e2eNamespace)
+		secondNamespace = mocks.PrepareNamespace(secondNamespaceName, map[string]string{e2eLabelKey: e2eLabelValue})
+		createResource(k8sClient, &secondNamespace)
+	})
+
+	Context("Validate get Namespaces route", func() {
+		It("Should get namespaces", func() {
+			baseURI := fmt.Sprintf("%s/v1/%s", platformURL, testutils.NamespaceKey)
+			status, response := performAuthorizedHTTPRequest(httpClient, nil, http.MethodGet, baseURI, "", "", userToken)
+
+			expectedResponse := map[string]interface{}{
+				testutils.CountKey: 2,
+				testutils.NamespaceKey: []map[string]interface{}{
+					{
+						testutils.NameKey: oneNamespaceName,
+					},
+					{
+						testutils.NameKey: secondNamespaceName,
+					},
+				},
+			}
+
+			Expect(status).Should(Equal(http.StatusOK))
+			Expect(expectedResponse[testutils.CountKey]).To(BeNumerically("<=", response[testutils.CountKey]))
```

Comment:
why not equal ?

Reply:
Since the tests run in parallel, additional namespaces are created so there's no guarantee that the `get` would only give me 2 namespaces

Reply:
The namespaces are cleaned up at the end of the tests to save time